jetstream/config.py

```diff
@@ -114,45 +115,76 @@ class ParameterDefinition:
     """
 
     name: str  # implicit in configuration
-    friendly_name: str
+    friendly_name: Optional[str] = None
     description: Optional[str] = None
-    value: Optional[str] = None
+    value: Optional[Union[str, Dict[str, str]]] = None
     distinct_by_branch: Optional[bool] = False
     default: Optional[Union[str, Dict[str, Any]]] = None
 
-    @classmethod
-    def from_dict(cls, d: Mapping[str, Any]) -> "ParameterDefinition":
-        definition = {
-            "name": d["name"],
-            "friendly_name": d.get("friendly_name"),
-            "description": d.get("description"),
-            "value": d.get("value"),
-            "distinct_by_branch": d.get("distinct_by_branch"),
-            "default": d.get("default"),
-        }
+    def validate(self) -> "ParameterDefinition":
+        """
+        Validates that branch related configuration is correct.
+
+        It should not run on every instance of ParameterDefinition object.
+        Outcome parameters containing defaults would not always adhere to
+        the rules defined here.
+        """
+
+        if self.distinct_by_branch and not (
+            isinstance(self.value, dict) or isinstance(self.default, dict)
+        ):
+            error_msg = (
+                f"Parameter {self.name} configured "
+                "to be distinct by branch, a mapping expected in the following format: "
+                'for values: value.branch_1 = "1"'
+                'and defaults: default.branch_1 = "1"'
```

Comment:
We could even link to the Experimenter docs for how to configure these parameters here.

```suggestion
                f"Parameter {self.name} configured "
                "to be distinct by branch, a mapping expected in the following format: "
                'for values: value.branch_1 = "1"'
                'and defaults: default.branch_1 = "1"'
                'See https://experimenter.info/jetstream/outcomes#parameterizing-outcomes for more information'
```